File: NSURLSessionSample/NSURLSessionSample/Classes/ViewController.m
//
//  ViewController.m
//  NSURLSessionSample
//
//  Created by Saurav Nagpal on 17/11/14.
//  Copyright (c) 2014 Saurav Nagpal. All rights reserved.
//

#import "UIImage+Downlading.h"
#import "HTTPSession.h"
#import "ViewController.h"
#define imageUrl    @"http://1.bp.blogspot.com/-N2fSe2JF0rg/T-vAeJcx4tI/AAAAAAAAErU/bOy8ya936xA/s1600/Tiger+3D+Wallpapers.jpg"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    /*HTTPSession* httpSession = [[HTTPSession alloc] init];
    [httpSession startDownloadingDataWithUrlString:imageUrl];*/
    /*NSURL* url = [NSURL URLWithString:imageUrl];
    [UIImage imageDownloadWithUrl:url onCompletion:^(NSURL *location, NSURLResponse *response, NSError *error){
        
    }];*/
    [self performSelectorInBackground:@selector(downloadImage:) withObject:@"thread1"];
    NSThread* thread = [[NSThread alloc] initWithTarget:self selector:@selector(downloadImage:) object:@"thread2"];
    [thread start];
    [self performSelectorInBackground:@selector(downloadImage:) withObject:@"thread3"];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void) downloadImage:(NSString*)threadID{
    NSLog(@"EnterThread%@",threadID);
    @synchronized(self){
        NSURL* url = [NSURL URLWithString:imageUrl];
        UIImage* image = [UIImage imageWithData:[NSData dataWithContentsOfURL:url]];
        NSLog(@"thread%@",threadID);
    }
   
}

@end

File: FaceGestureSample/FaceGestureSample/Classes/Controller/UIViewController/ViewController.swift
//
//  ViewController.swift
//  FaceGestureSample
//
//  Created by Saurav Nagpal on 14/07/14.
//  Copyright (c) 2014 Saurav Nagpal. All rights reserved.
//

import UIKit

class ViewController: UIViewController, VideoCapturing, FaceDetecting {
    var videoCaptureSession     : VideoCaptureSession!
    var faceDetectionManager    :FaceDetectionManager!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func viewWillAppear(animated: Bool){
        super.viewWillAppear(animated);
        
        
    }
    
    override func viewDidAppear(animated: Bool){
        super.viewDidAppear(animated);
        if videoCaptureSession == nil {
            videoCaptureSession = VideoCaptureSession(delegate:self);
        }
        if faceDetectionManager == nil {
            faceDetectionManager = FaceDetectionManager(delegate: self)
        }
        
        videoCaptureSession.startSession();
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func videoCaptureSession(session:VideoCaptureSession, failWithError error:NSError?){
        if let actualError = error{
            Utility.showAlert(title: FaceDetectionConstant.errorTitle, withMessage: actualError.localizedDescription, viewCtrl: self, handler: nil);
            NSLog(actualError.localizedDescription)
        }
       
    }
    
    func videoCaptureSession(session:VideoCaptureSession, didCaptureImage image:CIImage!){
        //self.faceDetectionManager.detectFeatureFromImage(image)
    }
    
    func faceDetector(detetor:FaceDetectionManager, didDetectMovment:faceMovementTypeEnum){
        
    }
}

